### DISCUSSION TITLE
Welcome to bladecoder-adventure-engine Discussions!

### DISCUSSION THREAD

**Discussion by bladecoder** [bladecoder/bladecoder-adventure-engine#53] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pfaz69** [comment]
Hi,
great tool!

I'm wondering if with this engine is it possible to attach scripts to game characters to make them perform some specific actions?

As an example, think of a sentinel following a prescribed path with some randomness: normally the sentinel stands; randomly, he takes a walk and goes way and back from his post (for instance he goes from left to right and then back to left to guard a gate); if something "suspicious" happens during his walk he freezes and points his rifle. 

I'm thinking that this would be possible if a path for him can be prescribed (as it happens for the player's avatar) and a script (or a .java file) can be attached to the sentinel.

Is this possible or is there another way to generate this kind of behaviors?

Many thanks for any clarification,

Paolo.
